dask/dataframe/core.py

```diff
@@ -5153,6 +5155,9 @@ def map_partitions(
     """
     name = kwargs.pop("token", None)
 
+    if has_keyword(func, "partition_info"):
+        kwargs["partition_info"] = {"number": -1, "divisions": None}
+
```

Comment:
You need to set `partition_info` to a valid dict before passing runnning the function on `meta` which happens below in `meta = _emulate(...`.

Reply:
See https://github.com/dask/dask/pull/6755/commits/32318b0d26c2a9d8363af48fbaeff355d7882b7f for instance

Reply:
Thank you @jsignell

Reply:
I saw his code and wasn't sure why `partition_info` was being initialized to 2 different values on different lines. This explains it.